CONTRIBUTORS.md

```diff
@@ -0,0 +1,105 @@
+
+## Issue Allocation
+Each issue someone is currently working on should have an assignee. If you want to contribute to an issue someone else is already working on please make sure to get in contact with that person via slack or github and organize yourself.
+
+If you want to work on an open issue, please post a comment telling that you will work on that issue, we will assign you ass the assignee then.
+
+**Caution**: We try our best to keep the assignee up-to-date but as we are all humans with our own schedule delays are possible, so make sure to check the comments once before you start working on an issue even when noone is assigned to it.
+
+
+## Writing Test Cases
+Always make sure to create the necessary tests and keep test coverage at 100%. You can always ask for help in slack or via github if you don't feel confidant about your tests.
+
+We aim to have a 100% test coverage, and the Travis CI will fail if the coverage is below this value. You can evaluate your coverage using the following commands.
+```
+coverage run --omit=*/venv/*,setup.py,.eggs/* setup.py test
+coverage report --fail-under 100 -m
+```
+
+Pysyft is using `pytest` to execute the test cases.
+
+#### Parametrize your Test Cases
+Sometimes you want to test functions that hold multiple arguments, which again can have multiple values. To test this, please parametrize your tests.
+
+Example:
+```
+@pytest.mark.parametrize(
+        "compress, compressScheme", [(True, "lz4"), (False, "lz4"), (True, "zstd"), (False, "zstd")]
+    )
+def test_hooked_tensor(self, compress, compressScheme):
+    TorchHook(torch)
+
+    t = Tensor(numpy.random.random((100, 100)))
+    t_serialized = serialize(t, compress=compress, compressScheme=compressScheme)
+    t_serialized_deserialized = deserialize(
+        t_serialized, compressed=compress, compressScheme=compressScheme
+        )
+    assert (t == t_serialized_deserialized).all()
+```
+
+## Documentation and Codestyle
+To ensure code quality and make sure other people can understand your changes, you have to document your code. For documentation we are using the Google Python Style Rules which can be found [here](https://github.com/google/styleguide/blob/gh-pages/pyguide.md). A well wrote example can we viewed [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+
+You documentation should not describe the obvious, but explain what's the intention behind the code and how you tried to realize your intention.
+
+You should also document non self-explanatory code fragments e.g. complicated for-loops. Again please do not just describe what each line is doing but also explain the idea behind the code fragment and why you decided to use that exact solution.
+
+#### Imports
+For better merge compatibility each import is within a separate line. Mulitple imports from one package are written in one line each.
+
+Example:
+```
```

Comment:
Maybe also put something about relative imports

Reply:
Sure! Good idea, do you have a suggestion what should be added?